service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt

```diff
@@ -396,30 +397,46 @@ fun sendPlacementToVarda(client: VardaClient, vardaDecisionId: Long, evakaServic
     return res.vardaPlacementId
 }
 
+private fun guardiansLiveInSameAddress(guardians: List<VardaGuardianWithId>): Boolean {
+    val validResidenceCodes = guardians.map { g -> g.asuinpaikantunnus }
+        .filterNotNull().filter { asuinpaikantunnus -> asuinpaikantunnus.length > 0 }
+    return validResidenceCodes.size == guardians.size && validResidenceCodes.toSet().size == 1
+}
+
 fun sendFeeDataToVarda(vardaClient: VardaClient, db: Database.Connection, newVardaServiceNeed: VardaServiceNeed, evakaServiceNeed: EvakaServiceNeedInfoForVarda, feeDataByServiceNeed: FeeDataByServiceNeed): List<Long> {
-    val feeResponseIds = feeDataByServiceNeed.feeDecisionIds.map { feeId ->
+    val guardians = getChildVardaGuardians(db, evakaServiceNeed.childId)
+    check(guardians.isNotEmpty()) { error("VardaUpdate: could not create fee data for ${evakaServiceNeed.id}: child has no guardians") }
```

Comment:
I don't know if it's even possible for a child to not have a guardian, but I feel like it shouldn't be an error. Maybe just log and return?